runtime/devnet/src/config/api/versioning.rs

```diff
@@ -159,15 +161,72 @@ impl From<DispatchError> for V0Error {
 
 #[cfg(test)]
 mod tests {
+	use frame_system::Call;
 	use pop_primitives::{ArithmeticError::*, Error, TokenError::*, TransactionalError::*};
 	use sp_runtime::ModuleError;
 	use DispatchError::*;
 
 	use super::*;
 
+	#[test]
+	fn from_versioned_runtime_call_to_runtime_call_works() {
+		let call =
+			RuntimeCall::System(Call::remark_with_event { remark: "pop".as_bytes().to_vec() });
+		assert_eq!(RuntimeCall::from(VersionedRuntimeCall::V0(call.clone())), call);
+	}
+
+	#[test]
+	fn from_versioned_runtime_read_to_runtime_read_works() {
+		let read = RuntimeRead::Fungibles(fungibles::Read::<Runtime>::TotalSupply(42));
+		assert_eq!(RuntimeRead::from(VersionedRuntimeRead::V0(read.clone())), read);
+	}
+
+	#[test]
+	fn from_versioned_runtime_result_works() {
+		let result = RuntimeResult::Fungibles(fungibles::ReadResult::<Runtime>::TotalSupply(1_000));
+		let v0 = 0;
+		let v1 = 1;
+		assert_eq!(
+			VersionedRuntimeResult::try_from((result.clone(), v0)),
+			Ok(VersionedRuntimeResult::V0(result.clone()))
+		);
+		// Unknown version.
+		assert_eq!(
+			VersionedRuntimeResult::try_from((result.clone(), v1)),
+			Err(pallet_contracts::Error::<Runtime>::DecodingFailed.into())
+		);
+	}
+
+	#[test]
+	fn from_versioned_runtime_result_to_bytes_works() {
+		let result = RuntimeResult::Fungibles(fungibles::ReadResult::<Runtime>::TotalSupply(1_000));
+		assert_eq!(<Vec<u8>>::from(VersionedRuntimeResult::V0(result.clone())), result.encode());
+	}
+
+	#[test]
+	fn from_versioned_error_works() {
```

Comment:
```suggestion
	fn versioned_error_works() {
```
We are converting into a versioned error from a current dispatch error and a version identifier.